source/dotnet/Test/AdaptiveCards.Test/SerializationTests.cs

```diff
@@ -48,6 +48,65 @@ public void TestCardsSerializeInTheCorrectOrder()
             
         }
 
+
+
+        [TestMethod]
+        public void TestSkippingUnknownElements()
+        {
+            var json = @"{
+  ""type"": ""AdaptiveCard"",
+  ""version"": ""1.0"",
+  ""body"": [
+    {
+      ""type"": ""IDunno"",
+      ""text"": ""Hello""
+    },
+    {
+      ""type"": ""TextBlock"",
+      ""text"": ""Hello""
+    }
+  ],
+  ""actions"": [
+    {
+      ""type"": ""Action.IDunno"",
+      ""title"": ""Action 1""
+    },
+    {
+      ""type"": ""Action.Submit"",
+      ""title"": ""Action 1""
+    }
+  ]
+}";
+            
+            var result = AdaptiveCard.FromJson(json);
+
+            Assert.IsNotNull(result.Card);
+            Assert.AreEqual(1, result.Card.Body.Count);
+            Assert.AreEqual(1, result.Card.Actions.Count);
+        }
+
+
+        [TestMethod]
+        public void Test_MissingTypeSkipsElement()
+        {
+            var json = @"{
+  ""type"": ""AdaptiveCard"",
+  ""version"": ""1.0"",
+  ""body"": [
+    {
+      ""type"": ""TextBlock"",
+      ""text"": ""Hello""
+    },
+    {
+      ""text"": ""Hello""
+    }
+  ]
+}";
+
+            var result = AdaptiveCard.FromJson(json);
+            Assert.AreEqual(1, result.Card.Body.Count);
```

Comment:
Aren't we supposed to fail here? I thought `type` is a required property on all elements, and therefore the schema is invalid (not just warning but invalid)

Reply:
Yea I think that's actually correct. I'll update the test and add this to our sample test scenarios